src/wp-includes/class-wp-block-type.php

```diff
@@ -295,15 +295,19 @@ public function __construct( $block_type, $args = array() ) {
 	 *
 	 * @param string $name Deprecated property name.
 	 *
-	 * @return string|null|void The value read from the new property if the first item in the array provided,
-	 *                          null when value not found, or void when unknown property name provided.
+	 * @return string|string[]|null|void The value read from the new property if the first item in the array provided,
+	 *                                   null when value not found, or void when unknown property name provided.
 	 */
 	public function __get( $name ) {
 		if ( ! in_array( $name, $this->deprecated_properties ) ) {
 			return;
 		}
 
 		$new_name = $name . '_handles';
+
+		if ( count( $this->{$new_name} ) > 1 ) {
```

Comment:
Could we use `property_exists` here?

Reply:
What if this is a string? How about a check for `is_array` instead of a count? Or even `is_iterable`

Reply:
So this one assumes that `$this->{$new_name}` exists and is an array (as those properties are [initialized as such](https://github.com/WordPress/wordpress-develop/blob/a7a9e7602c51785de5261c15d8a720c535e829bf/src/wp-includes/class-wp-block-type.php#L168-L206)), and checks if that array has more than one item. So we can't really replace it with `property_exists` 🤔 

Or do you mean an additional `property_exists` check to make sure `$this->{$new_name}` is there in the first place? 🤔 Is that needed? Requesting an object's property that _doesn't_ exist (e.g. `$obj->bogus`) is going to throw an error anyway, since it's the caller's responsibility to make sure that it _does_ exist, no? 🤔

Reply:
Blinding checking like this feels like is going to a problem. There are magic properties and other things that should not access. It is always good to check if a property exists before accessing like this, is a good idea.

Reply:
Specially with changes like this being added to PHP https://php.watch/versions/8.2/dynamic-properties-deprecated

Reply:
Gotcha, thanks for the pointer. I'll add the check 👍

Reply:
45775e0